WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersPeriodViewController.swift

```diff
@@ -0,0 +1,215 @@
+        childViewContrllers.forEach { childViewController in
+            addChild(childViewController)
+            childViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        }
+    }
+
+    func configureSubviews() {
+        view.addSubview(scrollView)
+        view.pinSubviewToSafeArea(scrollView)
+
+        scrollView.refreshControl = refreshControl
+
+        scrollView.addSubview(stackView)
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.pinSubviewToAllEdges(stackView)
+        NSLayoutConstraint.activate([
+            stackView.widthAnchor.constraint(equalTo: scrollView.widthAnchor)
+            ])
+
+        childViewContrllers.forEach { childViewController in
+            childViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        }
+
+        // Store stats.
+        let storeStatsPeriodView = storeStatsPeriodViewController.view!
+        stackView.addArrangedSubview(storeStatsPeriodView)
+        NSLayoutConstraint.activate([
+            storeStatsPeriodView.heightAnchor.constraint(equalToConstant: 380),
+            ])
+
+        // Top performers header.
+        let topPerformersHeaderView = TopPerformersSectionHeaderView(title:
+            NSLocalizedString("Top Performers",
+                              comment: "Header label for Top Performers section of My Store tab.")
+                .uppercased())
+        stackView.addArrangedSubview(topPerformersHeaderView)
+        let headerTopBorderView = createBorderView()
+        let headerBottomBorderView = createBorderView()
+        topPerformersHeaderView.addSubview(headerTopBorderView)
+        topPerformersHeaderView.addSubview(headerBottomBorderView)
+        NSLayoutConstraint.activate([
+            topPerformersHeaderView.heightAnchor.constraint(equalToConstant: 44),
+            // Top border view
+            headerTopBorderView.topAnchor.constraint(equalTo: topPerformersHeaderView.topAnchor),
+            headerTopBorderView.leadingAnchor.constraint(equalTo: topPerformersHeaderView.leadingAnchor),
+            headerTopBorderView.trailingAnchor.constraint(equalTo: topPerformersHeaderView.trailingAnchor),
+            // Bottom border view
+            headerBottomBorderView.bottomAnchor.constraint(equalTo: topPerformersHeaderView.bottomAnchor),
+            headerBottomBorderView.leadingAnchor.constraint(equalTo: topPerformersHeaderView.leadingAnchor),
+            headerBottomBorderView.trailingAnchor.constraint(equalTo: topPerformersHeaderView.trailingAnchor),
+            ])
+
+        // Top performers.
+        let topPerformersPeriodView = topPerformersPeriodViewController.view!
+        stackView.addArrangedSubview(topPerformersPeriodView)
+        NSLayoutConstraint.activate([
+            topPerformersPeriodView.heightAnchor.constraint(equalToConstant: 359.5),
```

Comment:
why `.5` in this constraint? Is this to account for the border line getting drawn in?

Reply:
I calculated the height from the v3 top performers height of 465px, deducting the following UI components that are in the vertical stack view (in `Dashboard.storyboard`):

header view (moved to a separate view `TopPerformersSectionHeaderView`): 55.5px
border: 1px
button bar view (moved to the top as the time range selector): 48px
border: 1px
——
105.5px

Then I got 359.5px for just the top performers data view